pkg/cbox/user/rest/rest.go

```diff
@@ -507,12 +497,8 @@ func (m *manager) IsInGroup(ctx context.Context, uid *userpb.UserId, group strin
 }
 
 func extractUID(u *userpb.User) (string, error) {
-	if u.Opaque != nil && u.Opaque.Map != nil {
-		if uidObj, ok := u.Opaque.Map["uid"]; ok {
-			if uidObj.Decoder == "plain" {
-				return string(uidObj.Value), nil
-			}
-		}
+	if u.UidNumber == 0 {
+		return "", errors.New("rest: could not retrieve UID from user")
 	}
-	return "", errors.New("rest: could not retrieve UID from user")
+	return fmt.Sprintf("%v", u.UidNumber), nil
```

Comment:
There are several parts of code similar to this, where the Opaque map was checked before returning a value. I chose to check if UidNumber and GidNumber are zeros. This only works if uid and gid can't be zeros, so I'd appreciate if someone can verify this.

Reply:
The (0, 0) values are reserved for root users, so we can assume that none of the ordinary users would have these. That being said, in cases where we're unable to retrieve the UID or GID, these might be treated as privileged users and granted access to resources they're not supposed to access. Can you find such instances in eosfs and add checks? If a user has these values, raise an error.

Reply:
It seems that functions in `eosfs` use `getUser` function for any user related operations, so i updated it to check for uid and gid in fabd746. Please let me know if that's what you had in mind.